server/sonar-web/src/main/js/app/components/nav/component/ComponentNavMeta.tsx

```diff
@@ -18,47 +18,70 @@
 import BranchStatus from '../../../../components/common/BranchStatus';
-import { Branch, Component } from '../../../types';
 import DateTimeFormatter from '../../../../components/intl/DateTimeFormatter';
+import Favorite from '../../../../components/controls/Favorite';
+import HomePageSelect from '../../../../components/controls/HomePageSelect';
 import Tooltip from '../../../../components/controls/Tooltip';
 import { isShortLivingBranch } from '../../../../helpers/branches';
 import { translate } from '../../../../helpers/l10n';
+import { getCurrentUser } from '../../../../store/rootReducer';
 
-interface Props {
+interface StateProps {
+  currentUser: CurrentUser;
+}
+
+interface Props extends StateProps {
   branch?: Branch;
   component: Component;
 }
 
-export default function ComponentNavMeta(props: Props) {
-  const shortBranch = props.branch && isShortLivingBranch(props.branch);
-  const showVersion = props.component.version && !shortBranch;
+export function ComponentNavMeta({ branch, component, currentUser }: Props) {
+  const shortBranch = branch && isShortLivingBranch(branch);
+  const mainBranch = !branch || branch.isMain;
 
   return (
     <div className="navbar-context-meta">
-      <ul className="list-inline">
-        {props.component.analysisDate && (
-          <li>
-            <DateTimeFormatter date={props.component.analysisDate} />
-          </li>
+      {component.analysisDate && (
+        <div className="spacer-left">
+          <DateTimeFormatter date={component.analysisDate} />
+        </div>
+      )}
+      {component.version &&
+        !shortBranch && (
+          <Tooltip overlay={`${translate('version')} ${component.version}`} mouseEnterDelay={0.5}>
+            <div className="spacer-left text-limited">
+              {translate('version')} {component.version}
+            </div>
+          </Tooltip>
         )}
-        {showVersion && (
-          <li>
-            <Tooltip
-              overlay={`${translate('version')} ${props.component.version}`}
-              mouseEnterDelay={0.5}>
-              <span className="text-limited">
-                {translate('version')} {props.component.version}
-              </span>
-            </Tooltip>
-          </li>
+      {isLoggedIn(currentUser) &&
+        mainBranch && (
+          <div className="navbar-context-meta-secondary">
+            <Favorite
+              component={component.key}
+              favorite={Boolean(component.isFavorite)}
+              qualifier={component.qualifier}
+            />
+            <HomePageSelect
+              className="spacer-left"
+              currentPage={{ type: HomePageType.Project, parameter: component.key }}
+            />
+          </div>
         )}
-      </ul>
       {shortBranch && (
-        <div className="navbar-context-meta-branch">
-          <BranchStatus branch={props.branch!} />
+        <div className="navbar-context-meta-secondary">
+          <BranchStatus branch={branch!} />
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Forbidden non null assertion ([typescript:S2966](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS2966))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-1128&issues=AWB6OR6KUyqLuEgueoe7&open=AWB6OR6KUyqLuEgueoe7)

[ISSUE_KEY]: AWB6OR6KUyqLuEgueoe7